commit 429bce31c68e9a8d779c0c8a8303799fc11df1d6
Author: Ravindu Deshan <droidcasts@protonmail.com>
Date:   Fri Mar 21 18:42:48 2025 +0530

    drivers/misc/tzdev/tz_deploy_tzar: fix startup.tzar inclusion
    
    Signed-off-by: Denis Efremov <efremov@linux.com>
    Signed-off-by: Ravindu Deshan <droidcasts@protonmail.com>

diff --git a/prebuilt-images/N/tzdev/Makefile b/prebuilt-images/N/tzdev/Makefile
index 3a713ff04..1f96c1656 100644
--- a/prebuilt-images/N/tzdev/Makefile
+++ b/prebuilt-images/N/tzdev/Makefile
@@ -1,6 +1,6 @@
-subdir-ccflags-y += -I$(srctree)/drivers/misc/tzdev/include
-subdir-ccflags-y += -I$(srctree)/drivers/misc/tzdev
-ccflags-$(CONFIG_TZDEV_DEPLOY_TZAR) += -D"_STR(s)=\#s" -D"KBUILD_SRC=_STR($(KBUILD_SRC))"
+subdir-ccflags-y += -I$(srctree)/$(src)/include
+subdir-ccflags-y += -I$(srctree)/$(src)
+ccflags-$(CONFIG_TZDEV_DEPLOY_TZAR) += -Wa,-I$(srctree)/$(src)
 
 obj-$(CONFIG_TZDEV)			+= lib/
 obj-$(CONFIG_TZDEV)			+= teec/
diff --git a/prebuilt-images/N/tzdev/tz_deploy_tzar.c b/prebuilt-images/N/tzdev/tz_deploy_tzar.c
index fc16e191c..5a280e4ad 100644
--- a/prebuilt-images/N/tzdev/tz_deploy_tzar.c
+++ b/prebuilt-images/N/tzdev/tz_deploy_tzar.c
@@ -33,7 +33,7 @@
 __asm__ (
   ".section .init.data,\"aw\"\n"
   "tzdev_tzar_begin:\n"
-  ".incbin \"" KBUILD_SRC "/drivers/misc/tzdev/startup.tzar\"\n"
+  ".incbin \"startup.tzar\"\n"
   "tzdev_tzar_end:\n"
   ".previous\n"
 );
diff --git a/prebuilt-images/S/tzdev/Makefile b/prebuilt-images/S/tzdev/Makefile
index d0f764b40..ac199989c 100755
--- a/prebuilt-images/S/tzdev/Makefile
+++ b/prebuilt-images/S/tzdev/Makefile
@@ -1,6 +1,6 @@
-subdir-ccflags-y += -I$(srctree)/drivers/misc/tzdev/include
-subdir-ccflags-y += -I$(srctree)/drivers/misc/tzdev
-ccflags-$(CONFIG_TZDEV_DEPLOY_TZAR) += -D"_STR(s)=\#s" -D"KBUILD_SRC=_STR($(KBUILD_SRC))"
+subdir-ccflags-y += -I$(srctree)/$(src)/include
+subdir-ccflags-y += -I$(srctree)/$(src)
+ccflags-$(CONFIG_TZDEV_DEPLOY_TZAR) += -Wa,-I$(srctree)/$(src)
 
 obj-$(CONFIG_TZDEV)			+= lib/
 obj-$(CONFIG_TZDEV)			+= teec/
diff --git a/prebuilt-images/S/tzdev/tz_deploy_tzar.c b/prebuilt-images/S/tzdev/tz_deploy_tzar.c
index fc16e191c..5a280e4ad 100755
--- a/prebuilt-images/S/tzdev/tz_deploy_tzar.c
+++ b/prebuilt-images/S/tzdev/tz_deploy_tzar.c
@@ -33,7 +33,7 @@
 __asm__ (
   ".section .init.data,\"aw\"\n"
   "tzdev_tzar_begin:\n"
-  ".incbin \"" KBUILD_SRC "/drivers/misc/tzdev/startup.tzar\"\n"
+  ".incbin \"startup.tzar\"\n"
   "tzdev_tzar_end:\n"
   ".previous\n"
 );
